nix_update/update.py

```diff
@@ -123,12 +128,22 @@ def update_version(
                     "Could not find a url in the derivations src attribute"
                 )
         version
-        new_version = fetch_latest_version(package.url, preference, version_regex)
+        if preference != VersionPreference.BRANCH:
+            branch = None
+        elif version == "branch":
+            # fallback
+            branch = "master"
+        else:
+            assert version.startswith("branch=")
```

Comment:
The assert is should never be false because otherwise `preference != VersionPreference.BRANCH` should have been triggered. It mainly makes `branch[7:]` a little more clear.